src/main.cc

```diff
@@ -40,11 +40,16 @@ int main(int argc, const char** argv) {
   CommandLine::Init(argc, argv);
   CommandLine::RegisterMode("uci", "(default) Act as UCI engine");
   CommandLine::RegisterMode("selfplay", "Play games with itself");
+  CommandLine::RegisterMode("benchmark", "Quick benchmark");
 
   if (CommandLine::ConsumeCommand("selfplay")) {
     // Selfplay mode.
     SelfPlayLoop loop;
     loop.RunLoop();
+  } else if (CommandLine::ConsumeCommand("benchmark")) {
+    // Selfplay mode.
+    EngineLoop loop;
```

Comment:
It seems that almost no of EngineLoop/EngineController functionality is not used here.
Maybe a separate class/separate file(or directory) from BenchmarkController?

Also `// Selfplay mode` comment doesn't look correct.

Reply:
I'll give it a go - my first effort required a lot of code duplication, but now that I have a working example I may be able to simplify it.